pandas/core/dtypes/cast.py

```diff
@@ -699,7 +699,9 @@ def _maybe_promote(dtype: np.dtype, fill_value=np.nan):
             dtype = np.dtype(np.object_)
 
         elif issubclass(dtype.type, np.integer):
-            if not np.can_cast(fill_value, dtype):
+            try:
+                np_can_hold_element(dtype, fill_value)
```

Comment:
Maybe we should right a wrapper function for this to make our own can cast.

Reply:
Good idea. Wrapped in a helper function